utils/saasherder.py

```diff
@@ -120,13 +122,16 @@ def populate_desired_state(self, ri):
                     cluster, namespace, resources = \
                         self._process_template(url, path, hash_length,
                                                target, parameters)
-                    # check images
-                    self._check_images(resources)
                     # add desired resources
                     for resource in resources:
                         resource_kind = resource['kind']
                         if resource_kind not in managed_resource_types:
                             continue
+                        # check images
+                        image_error = self._check_images(resource)
```

Comment:
A bit confusing that that `check_images()` returns `True` when there's a problem. Anyway, the fact that you're putting that into a variable called `image_error` mitigates that confusion here.